### PR TITLE
RDM-4135: Media Viewer integration

### PR DESCRIPTION
**Pull request by MKaramath**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-4135
https://tools.hmcts.net/jira/browse/RDM-5614

### Change description ###
Media viewer Integration with SSR Exclusion

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/shared/components/palette/document/media-viewer.component.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  mediaURL: string = '';
+  mediaFilename: string = '';
+  mediaContentType: string = '';
+
+  public constructor(private windowService: WindowService) {
+  }
+
+  ngOnInit() {
+    const localstorageMedia = this.windowService.getLocalStorage(MEDIA_VIEWER);
```
Comment (MSancaktutar) [inline src/shared/components/palette/document/media-viewer.component.ts]: localstorageMedia -->  localStorageMedia

File: src/shared/services/document-management/document-management.service.ts [hunk 2]
```diff
@@ -26,4 +29,30 @@ export class DocumentManagementService {
...
+    return JSON.stringify(mediaViewer);
+  }
+
+  getContentType(caseField: CaseField): string {
+    let fileExtension = '';
+    if (caseField.value && caseField.value.document_filename) {
+      fileExtension = caseField.value.document_filename
+        .slice(caseField.value.document_filename.lastIndexOf('.') + 1);
+    }
+    return (fileExtension == 'pdf') ? DocumentManagementService.PDF : DocumentManagementService.IMAGE;
```
Comment (MSancaktutar) [inline src/shared/services/document-management/document-management.service.ts]: Is it guaranteed that the type is image if it is not pdf? Couldn't it be a third type? This method seems to be problematic.

File: src/shared/services/document-management/document-management.service.ts [hunk 3]
```diff
@@ -26,4 +29,30 @@ export class DocumentManagementService {
...
+      }
+    }
+    return JSON.stringify(mediaViewer);
+  }
+
+  getContentType(caseField: CaseField): string {
+    let fileExtension = '';
+    if (caseField.value && caseField.value.document_filename) {
+      fileExtension = caseField.value.document_filename
+        .slice(caseField.value.document_filename.lastIndexOf('.') + 1);
```
Comment (MSancaktutar) [inline src/shared/services/document-management/document-management.service.ts]: what if the entire filename is just an extension like "pdf"?

File: src/shared/services/document-management/document-management.service.spec.ts [hunk 4]
```diff
@@ -63,4 +65,51 @@ describe('DocumentManagementService', () => {
         .subscribe(documentMetadata => expect(documentMetadata).toEqual(RESPONSE));
     });
   });
+
+  describe('Media viewer', () => {
```
Comment (MSancaktutar) [inline src/shared/services/document-management/document-management.service.spec.ts]: We should have more tests for file names of specific cases. ccd may be allowing only certain types of files but the mediaviewer logic should cover all file types appropriately. The viewer itself, then, may be unable to render some file types, but it should be handled inside the viewer. Our file type extraction logic must not assume certain types only.

Comment (tzarsmango) [reply src/shared/services/document-management/document-management.service.spec.ts]: +1 should test for unexpected file types handling as well

Comment (tzarsmango) [reply src/shared/services/document-management/document-management.service.spec.ts]: ok, so doc extension addresses other types ;)

File: src/shared/components/palette/document/media-viewer.component.ts (hunk 1 above)
Comment (MKaramath) [reply src/shared/components/palette/document/media-viewer.component.ts]: Done.

File: src/shared/services/document-management/document-management.service.ts (hunk 2 above)
Comment (MKaramath) [reply src/shared/services/document-management/document-management.service.ts]: Refactored by introducing an array with the list of images

File: src/shared/services/document-management/document-management.service.spec.ts (hunk 4 above)
Comment (MKaramath) [reply src/shared/services/document-management/document-management.service.spec.ts]: Added more test cases.

**Comment by MSancaktutar** [comment]
merging to generate prerelease

**Pull request closed without merging**